mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/topnews/TopNewsAdapter.java

```diff
@@ -23,33 +22,47 @@
 /**
  * Copyright © Cliqz 2018
  */
-public class TopNewsAdapter extends RecyclerView.Adapter<TopNewsCard> implements RecyclerViewClickSupport.OnItemClickListener {
+public class TopNewsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>
+        implements RecyclerViewClickSupport.OnItemClickListener {
+
+    private static final int TYPE_HEADER = 0;
+    private static final int TYPE_NEWS = 1;
+
+    // Number of items to show in the 'Top News' list when the list is collapsed.
+    private static final int COLLAPSED_NUM_NEWS_COUNT = 0;
+    // By default, the 'Top News' list is not expanded i.e. it shows not more than
+    // COLLAPSED_NUM_NEWS_COUNT' items.
+    private static final boolean PREFS_DEFAULT_NEWS_VIEW_EXPANDED = false;
 
     private final Context context;
     private final HomePager.OnUrlOpenListener onUrlOpenListener;
-    private List<TopNews> topNews;
+
+    private final List<TopNews> topNews;
     private int numShowedNews = 0;
+    private final SharedPreferences sharedPreferences;
+    private boolean isNewsExpanded;
 
     public TopNewsAdapter(Context context, final HomePager.OnUrlOpenListener onUrlOpenListener) {
         this.context = context;
         this.onUrlOpenListener = onUrlOpenListener;
+        this.sharedPreferences = GeckoSharedPrefs.forProfile(context);
```

Comment:
Please use PreferenceManager class instead, and put your logic inside a function there.
it's a Singelton pattern instead of creating new sharedPreference objects.

Reply:
Did that. Thanks!